### PR TITLE
Create Group component

### LINKED ISSUE #1812: Create Group component
Based on #394 we need a "Button group" component. Be it for only for internal and/or external use we need a way of grouping several elements together with a distinct styling to show they are related to one another.

The Button group component should be a simple container component that groups its children and only applies rounded corner to the outer edges.

Notes:
* Container which removes border-radius of its children
* Only applies rounded corners to its outer edges
* No spacing between its children
* Styling of children should work for all elements
* Naming of said component, discuss what makes most sense
  * `Group` ~`ButtonGroup` or  (sub-component)`Button.Group`~

### PR DESCRIPTION
**Pull request by laliq**
Resolves #1812

### CODE REVIEW

**Comment by laliq** [comment]
I have created `ButtonGroup` in `Button.stories` just to have overview.

File: libraries/core-react/src/components/Button/Button.stories.tsx [hunk 1]
```diff
@@ -17,6 +19,12 @@ const FullWidthWrapper = styled.div`
   grid-gap: 16px;
 `
 
+const GroupWrapper = styled.div`
+  margin: 32px;
+  display: grid;
+  grid-template-columns: repeat(4, fit-content(100%));
```
Comment (mimarz) [inline libraries/core-react/src/components/Button/Button.stories.tsx:25]: Don't think we need this css? Maybe you can extend the `Wrapper` with styled and set `grid-gap: 0` if you need that margin for the story,

File: libraries/core-react/src/components/Button/Button.stories.tsx [hunk 2]
```diff
@@ -314,18 +322,27 @@ export const FullWidth: Story<ButtonProps> = () => (
...
+        <Button>Contained</Button>
+        <Button variant="outlined">Outlined</Button>
+        <Button variant="ghost">Ghost</Button>
+        <Button variant="ghost_icon">
+          <Icon data={menu} title="Ghost icon menu"></Icon>
+        </Button>
+      </Wrapper>
+    </EdsProvider>
+  )
+}
```
Comment (mimarz) [inline libraries/core-react/src/components/Button/Button.stories.tsx:345]: This should not be a changed in this PR or was this intentional?

File: libraries/core-react/src/components/Group/Group.tsx [hunk 3]
```diff
@@ -0,0 +1,36 @@
+import { forwardRef } from 'react'
+import styled from 'styled-components'
+import { group as tokens } from './Group.tokens'
+
+const { border } = tokens
+
+const GroupBase = styled.div`
```
Comment (mimarz) [inline libraries/core-react/src/components/Group/Group.tsx:7]: Suggest changing this to use the function style (`styled.div(() => css``)`), then you only need to do the `border.type === 'border' && border.radius` check once as a const.

File: libraries/core-react/src/components/Group/Group.tsx [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+  }
+`
+
+export type GroupProps = React.HTMLAttributes<HTMLDivElement>
+
+export const Group = forwardRef<HTMLDivElement, GroupProps>(function Group(
+  { children },
+  ref,
+) {
+  return <GroupBase ref={ref}>{children}</GroupBase>
```
Comment (mimarz) [inline libraries/core-react/src/components/Group/Group.tsx:35]: We should still spread the `...rest` on the `GroupBase` to conform with our other components, even when its just internal :)

**Review by mimarz — CHANGES_REQUESTED** [review]
Great work! 

Some small suggestions on cleaner code :)

File: libraries/core-react/src/components/Button/Button.stories.tsx (hunk 2 above)
Comment (laliq) [reply libraries/core-react/src/components/Button/Button.stories.tsx:345]: This was related to the conflict. I performed rebased, now everything is in check.

File: libraries/core-react/src/components/Button/Button.stories.tsx (hunk 1 above)
Comment (laliq) [reply libraries/core-react/src/components/Button/Button.stories.tsx:25]: Great, thanks

File: libraries/core-react/src/components/Button/Button.stories.tsx (hunk 2 above)
Comment (mimarz) [reply libraries/core-react/src/components/Button/Button.stories.tsx:339]: @laliq Are you sure you pulled the latest changes from develop locally before rebasing it onto this branch? We are missing from develop the past 5 days 🤔

Comment (laliq) [reply libraries/core-react/src/components/Button/Button.stories.tsx:339]: running merge now

Comment (laliq) [reply libraries/core-react/src/components/Button/Button.stories.tsx:339]: @mimarz looks good to me now

Comment (mimarz) [reply libraries/core-react/src/components/Button/Button.stories.tsx:339]: Now you have 73 files flagged as changed... 🙈

Comment (laliq) [reply libraries/core-react/src/components/Button/Button.stories.tsx:339]: I did this...
```
git checkout my-brunch
git merge origin/master
```
I guess it was wrong 😑

Comment (mimarz) [reply libraries/core-react/src/components/Button/Button.stories.tsx]:
```sh
git rebase develop
git push -f
```

**Review by mimarz — APPROVED** [review]
Great work! 👏 👍

Have a merry christmas and a happy new year! 🎅 🎄 ✨

**Pull request merged**